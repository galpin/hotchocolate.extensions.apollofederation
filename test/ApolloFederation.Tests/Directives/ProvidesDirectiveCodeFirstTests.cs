using System.Threading.Tasks;
using HotChocolate.Types;
using Microsoft.Extensions.DependencyInjection;
using Xunit;
using static HotChocolate.Extensions.ApolloFederation.Test;

namespace HotChocolate.Extensions.ApolloFederation.Directives;

public class ProvidesDirectiveCodeFirstTests
{
    [Fact]
    public async Task When_key_is_specified_on_object()
    {
        var schema = await BuildSchemaAsync(builder =>
        {
            builder.AddObjectType(x =>
            {
                x.Name("Review").Key("id");
                x.Field("id").Type<IntType>();
                x.Field("product").Type("Product").Provides("name");
            });
            builder.AddObjectType(x =>
            {
                x.Name("Product").Key("upc");
                x.Field("upc").Type<StringType>();
            });
            builder.AddQueryType();
        });

        var sut = schema.GetType<ObjectType>("Review");

        Assert.Collection(
            sut.Fields["product"].Directives,
            x => AssertEx.Directive(x, "provides", ("fields", "\"name\"")));
    }
}