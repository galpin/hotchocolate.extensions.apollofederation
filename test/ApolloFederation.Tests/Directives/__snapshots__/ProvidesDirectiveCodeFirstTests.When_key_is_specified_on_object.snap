{
  "data": {
    "_service": {
      "sdl": "type Product @key(fields: \u0022upc\u0022) {\n  upc: String\n}\n\ntype Review @key(fields: \u0022id\u0022) {\n  id: Int\n  product: Product @provides(fields: \u0022name\u0022)\n}"
    }
  }
}
